scripts/install-hooks.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -e
+
+HOOK_DIR=".git/hooks"
+SCRIPTS_DIR="scripts/"
+
+mkdir -p $HOOK_DIR
+mkdir -p $SCRIPTS_DIR
+
+cat > "$SCRIPTS_DIR/pre-push" << 'EOL'
+#!/bin/bash
+
+set -e
+
+echo "Running pre-push hook to check documentation..."
+
+if [[ -f "osdctl_commands.md" ]]; then
+    PRE_MD5=$(md5sum "osdctl_commands.md" | cut -d ' ' -f 1)
+else
+    PRE_MD5=""
+fi
+
+echo "Generating documentation..."
+if make generate-docs; then
+    echo "Documentation generated successfully."
+else
+    echo "Failed to generate documentation. Fix errors before pushing."
+    exit 1
+fi
+
+if [[ -f "osdctl_commands.md" ]]; then
+    POST_MD5=$(md5sum "osdctl_commands.md" | cut -d ' ' -f 1)
+    
+    if [[ "$PRE_MD5" != "$POST_MD5" ]]; then
+        echo "ERROR: Documentation is out of date. The pre-push hook updated it."
+        echo "Please add and commit the updated documentation before pushing:"
+        echo "    git add osdctl_commands.md"
+        echo "    git commit -m \"Update documentation\""
+        exit 1
+    else
+        echo "Documentation is up to date."
+    fi
+else
+    echo "ERROR: Documentation file osdctl_commands.md was not generated."
+    exit 1
+fi
```

Comment:
So I think you can greatly simplify the logic here by removing the state file altogether, and just generating the docs as part of the pre-commit hook. It should be pretty fast given it is Go generating markdown from the command structure. Then any diff will be caught and no need to check pre/post.